EpiAware/src/EpiAwareUtils/accumulate_scan.jl

```diff
@@ -0,0 +1,58 @@
+@doc raw"
+    Apply the `accumulate` function to the `AbstractAccumulationStep` object.
+    This is effectively a optimised version of a for loop that applies the
+    `AbstractAccumulationStep` object to the input data in a single pass.
+
+    # Arguments
+    - `acc_step::AbstractAccumulationStep: The accumulation step function.
+    - `initial_state`: The initial state of the accumulation.
+    - `ϵ_t::AbstractVector{<:Real}`: The input data.
+
+    # Returns
+    - `state::AbstractVector{<:Real}`: The accumulated state as returned by the
+    `get_state` function from the output of the `accumulate` function.
+
+    # Examples
+    ```julia
+    using EpiAware
+    struct TestStep <: AbstractAccumulationStep
+        a::Float64
+    end
+
+    function (step::TestStep)(state, ϵ)
+        new_state = step.a * ϵ
+        return new_state
+    end
+
+    acc_step = TestStep(0.5)
+    initial_state = zeros(3)
+
+    accumulate_scan(acc_step, initial_state, [1.0, 2.0, 3.0])
+
+    function get_state(acc_step::TestStep, initial_state, state)
+        return state
+    end
+
+    accumulate_scan(acc_step, initial_state, [1.0, 2.0, 3.0])
+    ```
+"
+function accumulate_scan(acc_step::AbstractAccumulationStep, initial_state, ϵ_t)
+    result = accumulate(acc_step, ϵ_t, init = initial_state)
```

Comment:
Very nice! Its nice use a `Base` function (i.e. `Base.accumulate`). 

I'm mentioning this for future consideration because it won't make much difference here for this but there is a lazy version `Iterators.accumulate` that might be handy in a future refactor (e.g. if we use SMC based inference in the future this could be a more efficient method of `accumulate`.

Reply:
Is there any reason not to do this now?

Reply:
f2f conclusion @SamuelBrand1  to make an issue